aopy/data/bmi3d.py

```diff
@@ -832,6 +835,46 @@ def get_target_locations(preproc_dir, subject, te_id, date, target_indices):
             raise ValueError(f"Target index {target_indices[i]} not found")
     return np.round(locations,4)
 
+def get_trajectory_frequencies(preproc_dir, subject, te_id, date):
+    # load task codes
+    config_dir = files('aopy').joinpath('config')
+    params_file = as_file(config_dir.joinpath('task_codes.yaml'))
+    with params_file as f:
+        task_codes = yaml_read(f)[0]
+
+    # load preproc data
+    data, metadata = load_preproc_exp_data(preproc_dir, subject, te_id, date)
+    params = json.loads(metadata['sequence_params'])
+    primes = np.asarray(list(sympy.primerange(0, sympy.prime(params['num_primes'])+1)))
+    even_idx = np.arange(len(primes))[0::2]
+    odd_idx = np.arange(len(primes))[1::2]
+    base_period = 20
+
+    # recreate random trial order of reference & disturbance frequencies
+    np.random.seed(params['seed'])
+    order = np.random.choice([0,1])
+    if order == 0:
+        trial_r_idx = np.array([even_idx, odd_idx]*params['ntrials'], dtype='object')
+        trial_d_idx = np.array([odd_idx, even_idx]*params['ntrials'], dtype='object')
+    elif order == 1:
+        trial_r_idx = np.array([odd_idx, even_idx]*params['ntrials'], dtype='object')
+        trial_d_idx = np.array([even_idx, odd_idx]*params['ntrials'], dtype='object')
+        
+    # get trajectory generator index used for each trial
+    trial_start_codes = [task_codes['CENTER_TARGET_ON']]
+    trial_end_codes = [task_codes['REWARD'], task_codes['OTHER_PENALTY'], 
+                       task_codes['HOLD_PENALTY'], task_codes['TIMEOUT_PENALTY']] # using TRIAL_END may not pass assertion
+    gen_trajectories, _ = get_kinematic_segments(preproc_dir, subject, te_id, date, 
```

Comment:
i think it might make more sense to use the get_trial_segments() function to segment the `data['task']['gen_idx']` vector into trials using `data['bmi3d_events']['time']` and `data['bmi3d_events']['code'] rather than using real-time get_kinematic_segments() wrapper, which uses `data['events']['timestamp']` and `data['events']['code']` recorded by the ecube. That way the events are the same, and therefore the segments will be the same, but there's no interpolation.